app/scripts/controllers/serviceInstances.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  .controller('ServiceInstancesController', function ($scope,
+                                                      $filter,
+                                                      $routeParams,
+                                                      APIService,
+                                                      BindingService,
+                                                      Constants,
+                                                      DataService,
+                                                      LabelFilter,
+                                                      Logger,
+                                                      ProjectsService) {
+    $scope.alerts = {};
+    $scope.applicationsByBinding = {};
+    $scope.bindings = {};
+    $scope.bindableServiceInstances = {};
+    //$scope.bindingsByApplicationUID = {};
+    $scope.bindingsByInstanceRef = {};
+    $scope.emptyMessage = "Loading...";
+    $scope.labelSuggestions = {};
+    $scope.projectName = $routeParams.project;
+    $scope.serviceClasses = {};
+    $scope.serviceInstances = {};
+    $scope.unfilteredServiceInstances = {};
+
+    var watches = [];
+
+    var updateFilter = function() {
+      $scope.serviceInstances = LabelFilter.getLabelSelector().select($scope.unfilteredServiceInstances);
+    };
+
+    var sortServiceInstances = function() {
+      $scope.bindableServiceInstances = BindingService.filterBindableServiceInstances($scope.unfilteredServiceInstances, $scope.serviceClasses);
+      $scope.unfilteredServiceInstances = BindingService.sortServiceInstances($scope.unfilteredServiceInstances, $scope.serviceClasses);
+    };
+
+    var groupBindings = function() {
+      // Build two maps:
```

Comment:
I suggest for now just creating a map of bindings by service instance name. Then you can show those in the table. We'll eventually want to show application name, but we can't do that until pod presets are enabled in a later release.

@jeff-phillips-18 FYI